README.md

```diff
@@ -1 +1,73 @@
+
+In addition, our Auth SDK is extemely complex and hard to maintain and extend due to the complexity in the state machine.
+
+### Goal
+
+The goal of this project is to automatically generate SDKs for Nhost based on the API schema. This will ensure that the SDKs are always up-to-date and consistent with the API.
+
+In addition, we want to make sure that our SDKs are easy to use and understand and that can be extended easily to accomodate different environments. This is specially important in the Javascript/Typescript SDKs as those need to run across a variety of environments (Node.js, React Native, Browser).
+
+## Objectives
+
+### Languages
+
+To begin with we want to start supporting the following languages:
+
+- Javascript/Typescript
+- Go
+- Dart
+
+As the project progresses, we can add support for more languages but to keep things simple and focused we will start with these three. The variety in languages will help us understand the different challenges and requirements to support different languages.
+
+### Requirements
+
+- The SDKs should be automatically generated based on the API schema
+- The SDKs should be easy to use and understand and include documentation generated from the API schema
+- The SDKs should be consistent with the API
+- The SDKs should be easy to extend, specially when it relates to managing the session and authentication
+- The SDKs should be idiomatic to the language they are written for
+
+## Approach
+
+### Code Generation
+
+We will start by testing different code generation tools to generate the SDKs and ensuring the code generated supports the requirements outlined above and the code generated correctly reflects the API schema.
+
+### Examples
+
+We will design an example to test each generated SDK and ensure it meets the requirements. This will help us understand how the generated SDK feels, how easy it is to use and understand, and how easy it is to extend.
+
+### Community
+
+We want to involve the community in the development of the SDKs. This will ensure that the SDKs are easy to use and understand and that they cover as many use cases as possible. In the end, the SDKs are for the community and, thus, should be developed with the community.
+
+## Contributing
+
+We are currently looking for code generation tools that can help us generate the SDKs. If you have experience with code generation tools and would like to propose one, please open an issue with the details of the tool and why you think it would be a good fit for this project. If you see a tool that has been proposed and you have experience with it, please, don't hesitate to share your experience and thoughts on the tool.
+
+As we generate some sample SDKs with the proposed tools we will create issues on GitHub asking for help writing examples and asking for feedback on the generated code and the examples themselves. If you have a different environment from the ones we are testing, don't hesitate to open an issue to let us know.
```

Comment:
The API is the most important part IMO. 

I am particularly interested in the discussion around different environments (browser, node, etc) and the optional framework specific SDKs (react, vue, etc).